### PR TITLE
gitserver: use standardized actor propagation

### PR DESCRIPTION
**Pull request by bobheadxi**
Removes the custom actor propagation that `gitserver` and `gitserver.Client` uses, in favour of leveraging the standardized actor propagation introduced in https://github.com/sourcegraph/sourcegraph/issues/27918. There should be no behavioural changes, since this actor data is only used for observability on the `gitserver` side, and serverside backcompat with the old header is retained for the time being.

One particular usage of `gitserver.Client`, in `NewGitoliteSource`, required adding the actor transport manually because the external source constructors are provided a `httpcli.ExternalClientFactory`, which does not have the actor propagation transport set up.

Closes https://github.com/sourcegraph/sourcegraph/issues/28448

### CODE REVIEW

**Review by keegancsmith — APPROVED** [review]
LGTM. Just remember we don't roll out gitserver at the same rate as frontend. So when this auto rolls out our actor observability will be broken. This is not ideal, so I'd recommend manually bumping gitserver or make this PR check both headers and then in future remove checking the old header.

File: cmd/gitserver/main.go [hunk 1]
```diff
@@ -139,9 +139,10 @@ func main() {
 	}
 
 	// Create Handler now since it also initializes state
-
 	// TODO: Why do we set server state as a side effect of creating our handler?
-	handler := ot.HTTPMiddleware(trace.HTTPTraceMiddleware(gitserver.Handler(), conf.DefaultClient()))
+	handler := gitserver.Handler()
+	handler = actor.HTTPMiddleware(handler)
+	handler = ot.HTTPMiddleware(trace.HTTPTraceMiddleware(handler, conf.DefaultClient()))
```
Comment (ryanslade) [inline cmd/gitserver/main.go:145]: nit: We should also rename to `trace.HTTPMiddleWare`

File: internal/repos/gitolite.go [hunk 2]
```diff
@@ -40,12 +39,12 @@ func NewGitoliteSource(svc *types.ExternalService, cf *httpcli.Factory) (*Gitoli
...
-	hc, err := cf.Doer(func(c *http.Client) error {
-		if tr, ok := c.Transport.(*http.Transport); ok {
-			tr.MaxIdleConnsPerHost = 500
-		}
-		return nil
-	})
+	gitserverDoer, err := cf.Doer(
+		httpcli.NewMaxIdleConnsPerHostOpt(500),
+		// The provided httpcli.Factory is one used for external services - however,
+		// GitoliteSource asks gitserver to gitolite instead, so we have to ensure
```
Comment (ryanslade) [inline internal/repos/gitolite.go:45]: Typo at `gitserver to gitlote`?

**Review by ryanslade — APPROVED** [review]
LGTM, agree with Keegan's concerns around deployment though

File: cmd/gitserver/main.go (hunk 1 above)
Comment (bobheadxi) [reply cmd/gitserver/main.go:145]: I'll open a follow-up!

File: internal/repos/gitolite.go (hunk 2 above)
Comment (bobheadxi) [reply internal/repos/gitolite.go]: ah thanks - fixed in 005bc890f72592eeca3efb050d6768e9b06f0af1

**Comment by bobheadxi** [comment]
I've opted to add backcompat on the serverside here: 456c71076fc91d800fc88007e795880ef3568525

File: cmd/gitserver/main.go (hunk 1 above)
Comment (bobheadxi) [reply cmd/gitserver/main.go:145]: https://github.com/sourcegraph/sourcegraph/pull/28490

**Pull request merged**